docker-compose.dev.yml

```diff
@@ -1,27 +1,82 @@
 services:
+  nginx:
+    image: nginx:latest
+    ports:
+      - 8080:8080
+    configs:
+      - source: nginx_config
+        target: /etc/nginx/nginx.conf
+    networks:
+      - public
+    depends_on:
+      server:
+        condition: service_healthy
+        restart: true
+
   server:
     build:
       context: ./server
       dockerfile: Dockerfile.dev
-    ports:
-      - '5001:5001'
+    # ports:
+    #   - 5001:5001
     volumes:
       - ./server:/app
       - /app/node_modules
     environment:
       - NODE_ENV=development
     env_file:
       - .env
+    networks:
+      - private
+      - public
+    depends_on:
+      db:
+        condition: service_healthy
+        restart: true
+    healthcheck:
```

Comment:
this is probably unnecessary for dev server

Reply:
I meant the healthcheck of dev server itself

Reply:
mentioned below